File: DSA/NiharikaChand_590017697/Day19Question2.java
import java.util.*;
class Day19Question2 
{
     public static int peak(int arr[], int n)
    {
        int l= 0, h = n-1;
        while(l < h)
        {
            int mid = l + (h - l) / 2;
            boolean left = (mid == 0 || arr[mid] > arr[mid-1]);
            boolean right = (mid== n- 1 || arr[mid] > arr[mid+1]);

            if(left && right)
            {
                return mid;
            }
            if(mid > 0 && arr[mid] < arr[mid-1])
            {
                h = mid -1;
            }   
            else
            {
                l = mid + 1;
            }
        }
        return 1;
    } 

   public static void main(String args[]) 
   {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter size of array: ");
        int n = sc.nextInt();
        int arr[] = new int[n]; int i;
        for(i=0; i<n; i++)
        {
            System.out.println("Enter an element: ");
            arr[i] = sc.nextInt();
        }
        System.out.println("\n" + peak(arr, n));
    }
}

File: DSA/Rahul_Mankani_590014725/Day19_Question2.c
#include <stdio.h>

int main() 
{
    int arr[100];
    int n;
    int peak = -1;
    printf("Enter number of elements: ");
    scanf("%d", &n);
    if (n <= 0 || n > 100) 
    {
        printf("Invalid size! Enter between 1 and 100.\n");
        return 1;
    }
    printf("Enter elements: ");
    for (int i = 0; i < n; i++) 
    {
        scanf("%d", &arr[i]);
    }
    int left = 0;
    int right = n - 1;
    while (left <= right) 
    {
        int mid = (left + right) / 2;
        if ((mid == 0 || arr[mid] > arr[mid - 1]) &&
            (mid == n - 1 || arr[mid] > arr[mid + 1])) 
        {
            peak = mid;
            break;
        }
        else if (mid > 0 && arr[mid - 1] > arr[mid]) 
        {
            right = mid - 1;
        }
        else 
        {
            left = mid + 1;
        }
    }
    if (peak != -1) 
    {
        printf("Peak element %d found at index %d.\n", arr[peak], peak);
    } 
    else 
    {
        printf("No peak element found.\n");
    }
    return 0;
}
